library/core/src/slice/sort/shared/smallsort.rs

```diff
@@ -845,7 +846,21 @@ unsafe fn bidirectional_merge<T: FreezeMarker, F: FnMut(&T, &T) -> bool>(
 
 #[inline(never)]
 fn panic_on_ord_violation() -> ! {
-    panic!("Ord violation");
+    // This is indicative of a logic bug in the user-provided comparison function or Ord
+    // implementation. They are expected to implement a total order as explained in the Ord
+    // documentation.
+    //
+    // By raising this panic we inform the user, that they have a logic bug in their program. If a
+    // strict weak ordering is not given, the concept of comparison based sorting makes no sense.
+    // E.g.: a < b < c < a
```

Comment:
"makes no sense" isn't quite right, I think. This could be clearer about why we *choose* to not, instead, e.g. just allow this to happen and give back a somewhat-random permutation. The current algorithm would be memory-unsound if it tried to continue, yes? Would it even terminate? It would be inconsistent at best, right, even if we tried to force it to early-return, such that two `.sort()` calls would permute the list in different ways each time?

Basically, something like "We choose to not pay a zillion costs that would impose a burden on correct Ord impls. Instead, we bail if the comparison function is inconsistent".

Reply:
"they probably didn't mean it lol" is also fine, idc.

Reply:
Thanks for the detailed feedback! I'd say the primary motivation for panicking is to inform the user of a likely unintentional logic bug.

Reply:
Yeah, that seemed likely, I just noticed we are making an explicit choice between "do what I said" vs. "do what I mean", here.

It also will be even more justified, I think, by the fact that while `sort` and `sort_by` use the same code path, I _suspect_ we'll find out that `sort_by` hits this panic far more (because it gets more one-off functions that may be poorly tested).